programs/Makefile

```diff
@@ -107,9 +107,9 @@ hash/generic_sum$(EXEXT): hash/generic_sum.c $(DEP)
 
 # Unlike other programs, this one runs on the host even if cross-compiling,
 # and it does not link with mbed TLS libraries.
-host/frontend$(EXEXT): host/frontend.c $(DEP) host/frontend-config.h ../include/mbedtls/serialize.h
-	echo '  CC    host/frontend.c ../library/net_sockets.c'
-	$(CC) -I. $(LOCAL_CFLAGS) $(CFLAGS) host/frontend.c ../library/net_sockets.c $(LDFLAGS) -o $@
+host/frontend$(EXEXT): host/frontend.c $(DEP) host/frontend-config.h ../include/mbedtls/serialize.h ../include/mbedtls/fsio.h
+	echo '  CC    host/frontend.c ../library/net_sockets.c ../library/fsio.c'
+	$(CC) -I. $(LOCAL_CFLAGS) $(CFLAGS) -DMBEDTLS_CONFIG_FILE="\"host/frontend-config.h\"" host/frontend.c ../library/net_sockets.c ../library/fsio.c $(LDFLAGS) -o $@
```

Comment:
As discussed created Jira ticket IOTSSL-1852 to review this in more details outside this PR. Summary here:

- File frontend.c includes file frontend-config.h. 
- frontend-config.h is frontend.c counterpart of mbedtls/include/mbedtls/config.h. 
- frontend.c is linked with net_socket.c and fsio.c 
- For intended behaviour of net_socket.c and fsio.c they  should be compiled with same config as fronend.c.
- net_socket.c and fsio.c are also compiled in the Mbed TLS library. They include config file with following code:

{code:C}
#if !defined(MBEDTLS_CONFIG_FILE)
#include "config.h"
#else
#include MBEDTLS_CONFIG_FILE
#endif
{code}

- Based on above snippet one solution is to pass flag MBEDTLS_CONFIG_FILE while building. This is how it is done in this PR.
- This may not work as pointed by @gilles-peskine-arm  on VS and IAR.
- Alternative solution is to rename frontend-config.h to config.h and override header file lookup to pick this file.
- This may create issue with mbedtls/scripts/generate_errors.pl that checks that no two files have same name.
- This issue need to be looked upon in detail.
- Current implementation does not break anything as frontend.c is only compiled with gcc. Also abstract-fs tests are not part of build system hence a new platform will break due to this new file.